packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -177,26 +212,20 @@ const createManagerApi: CreateManagerApi = ({ onSubmit, clearOnUnmount, initiali
         }
       }
     };
-
-    // TODO: initial only first time -> have field states in global?
-    if (shouldExecute(initializeOnMount, field.initializeOnMount) || !Object.prototype.hasOwnProperty.call(state.values, field.name)) {
-      state.values[field.name] = field.value;
-    }
   }
 
   function unregisterField(field: Omit<FieldConfig, 'render'>): void {
     delete state.fieldListeners[field.name].fields[field.internalId];
 
     if (isLast(state.fieldListeners, field.name)) {
       state.registeredFields = state.registeredFields.filter((fieldName: string) => fieldName !== field.name);
-      delete state.fieldListeners[field.name];
 
       if (shouldExecute(clearOnUnmount, field.clearOnUnmount)) {
         state.values[field.name] = field.value;
       }
-    } else {
-      state.fieldListeners[field.name].count -= 1;
     }
+
+    state.fieldListeners[field.name].count -= 1;
```

Comment:
the `-=` and `+=` is according to some fairly buggy :man_shrugging:  (you can find a heaps of discussions around this topic). Can we switch to the long version? x = x + 1

Reply:
switched